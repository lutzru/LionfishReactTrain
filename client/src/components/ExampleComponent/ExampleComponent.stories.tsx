import { action } from '@storybook/addon-actions'
import { withKnobs } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/react'
import ExampleComponent, { ExampleComponentProps } from 'components/ExampleComponent/ExampleComponent'
import * as React from 'react'
import 'utils/ i18n'

const stories = storiesOf('Example Component', module)

const props: ExampleComponentProps = {
    currentValue: 0,
    decreaseValue: action('decreaseValue'),
    increaseValue: action('increaseValue'),
}

stories.addDecorator(withKnobs)
stories.add('default view', () => <ExampleComponent {...props} />)
