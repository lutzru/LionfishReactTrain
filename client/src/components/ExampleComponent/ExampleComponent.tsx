import { Button, Grid, Typography } from '@material-ui/core'
import React from 'react'
import TestIDs from 'utils/TestIDs'

export interface ExampleComponentProps {
    currentValue: number
    decreaseValue: () => void
    increaseValue: () => void
}

const ExampleComponent = (props: ExampleComponentProps): JSX.Element => {
    const { currentValue, decreaseValue, increaseValue } = props

    return (
        <Grid item container xs={12} spacing={3}>
            <Grid item xs={12}>
                <Typography {...TestIDs.GET_EXAMPLE('EXAMPLE_COUNTER')} className={'ExampleText'} color={'primary'}>
                    {currentValue}
                </Typography>
            </Grid>
            <Grid item xs={12} sm>
                <Button
                    {...TestIDs.GET_EXAMPLE('EXAMPLE_MINUS')}
                    onClick={decreaseValue}
                    fullWidth
                    variant={'outlined'}
                    color={'primary'}
                >
                    <Typography className={'ExampleText'}>-</Typography>
                </Button>
            </Grid>
            <Grid item xs={12} sm>
                <Button
                    {...TestIDs.GET_EXAMPLE('EXAMPLE_PLUS')}
                    onClick={increaseValue}
                    fullWidth
                    variant={'outlined'}
                    color={'primary'}
                >
                    <Typography className={'ExampleText'}>+</Typography>
                </Button>
            </Grid>
        </Grid>
    )
}

export default ExampleComponent
