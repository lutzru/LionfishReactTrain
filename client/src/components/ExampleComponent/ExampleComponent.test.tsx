import { Button } from '@material-ui/core'
import ExampleComponent, { ExampleComponentProps } from 'components/ExampleComponent/ExampleComponent'
import { shallow } from 'enzyme'
import * as React from 'react'
import * as renderer from 'react-test-renderer'

jest.mock('react-i18next', () => ({
    useTranslation: (): unknown => ({ t: (key: string): string => key }),
}))

const props: ExampleComponentProps = {
    currentValue: 0,
    decreaseValue: jest.fn(),
    increaseValue: jest.fn(),
}

describe('Example Component', () => {
    it('should render correctly', () => {
        const component = renderer.create(<ExampleComponent {...props} />)

        expect(component).toMatchSnapshot()
    })

    it('should handle clicks correctly', () => {
        const decreaseValue = jest.fn()
        const increaseValue = jest.fn()
        const component = shallow(
            <ExampleComponent {...props} decreaseValue={decreaseValue} increaseValue={increaseValue} />,
        )
        component.find(Button).at(0).simulate('click')
        component.find(Button).at(1).simulate('click')
        expect(decreaseValue).toHaveBeenCalledTimes(1)
        expect(increaseValue).toHaveBeenCalledTimes(1)
    })
})
