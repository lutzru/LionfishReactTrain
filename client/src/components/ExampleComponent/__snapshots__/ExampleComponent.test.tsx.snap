// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example Component should render correctly 1`] = `
<div
  className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-item MuiGrid-grid-xs-12"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
  >
    <p
      className="MuiTypography-root ExampleText MuiTypography-body1 MuiTypography-colorPrimary"
      data-testid="EXAMPLE_PAGE_EXAMPLE_COUNTER"
    >
      0
    </p>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-true"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-fullWidth"
      data-testid="EXAMPLE_PAGE_EXAMPLE_MINUS"
      disabled={false}
      onBlur={[Function]}
      onClick={[MockFunction]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <p
          className="MuiTypography-root ExampleText MuiTypography-body1"
        >
          -
        </p>
      </span>
    </button>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-true"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-fullWidth"
      data-testid="EXAMPLE_PAGE_EXAMPLE_PLUS"
      disabled={false}
      onBlur={[Function]}
      onClick={[MockFunction]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <p
          className="MuiTypography-root ExampleText MuiTypography-body1"
        >
          +
        </p>
      </span>
    </button>
  </div>
</div>
`;
